mirrord-agent/src/main.rs

```diff
@@ -58,6 +65,49 @@ impl Borrow<PeerID> for Peer {
     }
 }
 
+#[derive(Debug, Default)]
+struct FileManager {
+    pub open_files: HashMap<i32, File>,
+}
+
+impl FileManager {
+    // TODO(alex) [mid] 2022-05-19: Need to do a better job handling errors here (and in `read`)?
+    // Ideally we would send the error back in the `Response`, to be handled by the mirrod-layer
+    // side, and converted into some `libc::X` value.
+    pub(crate) fn open(&mut self, path: PathBuf) -> Result<i32> {
+        debug!("FileManager::open -> Trying to open file {path:#?}.");
+
+        // TODO(alex) [low] 2022-05-18: To avoid opening the same file multiple times, we could
+        // search `FileManager::open_files` for a path that matches the one passed as a parameter.
+        let file = std::fs::File::open(path)?;
```

Comment:
why do we keep a fd instead of just File object?

Reply:
okay that's weird it seems that you're creating a File struct then you transform it into fd then you convert it back to File? why?

Reply:
Hmm, I'm just grabbing the fd from the file to use as an id for `open_files`, not sure what you mean by "convert it back".

Reply:
Ah, got it. I would recommend against it, Imagine a case where the "remote fd" is 3 and then layer's `read` gets called with fd 3. This could either be the remote fd or it might be a socket fd that exists locally.
It's okay if you pass this id as the remote fd, but on layer you need the hashset to be mapped by a number that is guaranteed to not be taken as a fd, so maybe create a fd just to assure you're not reusing something that exists. (on the layer)